packages/flutter_tools/lib/src/compile.dart

```diff
@@ -250,6 +250,8 @@ class KernelCompiler {
         mainUri,
         newMainDart,
         mainFile,
+        // TODO(egarciad): Turn this on when the plugins are fixed.
```

Comment:
For those, we could send PRs to fix their pubspec.yaml.  Maybe, we could detect if the app being built is the plugin's example app. 

Also, this is an action item to the consumer of the plugin as well, since according to the design, a user can specify the default implementation of a plugin for a platform.